src/qrexec-agent/qrexec-agent.c

```diff
@@ -1075,9 +1171,10 @@ static DWORD WatchForEvents(HANDLE stopEvent)
             if (ERROR_SUCCESS != status)
             {
                 run = FALSE;
-                perror2(status, "HandleDaemonMessage");
+                win_perror2(status, "HandleDaemonMessage");
                 goto out;
             }
+            check_connections();
```

Comment:
Checking it here results in a funny effect - that port is released only after subsequent connection is made. So, repeated connections use port 513 (first available), then 514 (because 513 wasn't released _yet_) and then again 513 and so on.
For this case it isn't a big deal, but if you'd have windows-based DispVM, it would be destroyed with a delay (until subsequent qrexec call). So, a better solution is to include those handles in `WaitForMultipleObjects` call.

Reply:
Do you mean to extend waitObjects array with these handlers?

Reply:
yes

Reply:
fixed